### PR TITLE
Update gems, Allow Rails 5

### PR DESCRIPTION
**Pull request by calvinrachuy**
Rails 5 compatibility updates
  * ruby
  * activerecord
  * activesupport

Also resolves a security notice: https://github.com/Verba/serialized_settings/network/alert/Gemfile.lock/activerecord/open

This will be used in Collect. Compete instead uses a tagged version. (Connect does not use this.)

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,5 +1,4 @@
 language: ruby
 cache: bundler
 rvm:
-  - 2.1.2
-  - 1.9.3
+  - 2.3.7
```
Comment (micahbf) [inline .travis.yml]: collect is on 2.5, wonder if that would make more sense?

File: serialized_settings.gemspec [hunk 2]
```diff
@@ -8,11 +8,11 @@ Gem::Specification.new do |s|
...
-  s.add_dependency "activerecord", ">= 3.2", "< 5"
-  s.add_dependency "activesupport", ">= 3.2", "< 5"
+  s.add_dependency "activerecord"
+  s.add_dependency "activesupport"
 
   s.add_development_dependency "bump"
   s.add_development_dependency "rake"
   s.add_development_dependency "rspec"
-  s.add_development_dependency "sqlite3"
+  s.add_development_dependency "sqlite3", "~> 1.3.6"
```
Comment (micahbf) [inline serialized_settings.gemspec]: why's this pin needed?

Comment (calvinrachuy) [reply serialized_settings.gemspec]: It wouldn't bundle otherwise. https://github.com/rails/rails/issues/35153

Comment (calvinrachuy) [reply serialized_settings.gemspec]: I'll see if https://github.com/rails/rails/pull/35154 or related fixed it

Comment (calvinrachuy) [reply serialized_settings.gemspec]: it appears fixed. updated

File: .travis.yml (hunk 1 above)
Comment (calvinrachuy) [reply .travis.yml]: I held off because Compete is on 2.3. It's versioned, but will probably want to bump this soon enough.

I'm fine with either.

Comment (ndp) [reply .travis.yml]: I'm pretty sure you can just have lots of them here.

**Pull request merged**